### DISCUSSION TITLE
MarkdownV2 function unreliable

### DISCUSSION THREAD

**Discussion by tobya** [def-studio/telegraph#337] [General]
Perhaps I'm doing something wrong but send a chat using the markdownV2  function, often it will not be sent and no error is raised.

````
           $response =  $chat->markdownV2(
"Thank you for your message.  We have sent it via email to the office at the School and they will reply as soon as they can.
If it is an emergency, please phone Toby on +353 87 ... "
            )->send();

````

I dont get any error so I dont know what if anything is wrong with my text.  If I send this with ->markdown it works fine.

Am I doing something wrong?  Can I find out what the error is?

The $response varaible is

    {"cookies":null,"transferStats":null} 

ps. Thanks for a really great way to write a telegram bot.

**Comment by tobya** [comment]
Ok, I've figured out what everyone else probably knows.

 + needs to be escaped with \ as \+ 

However this does not always work and this text still wont sent.


````
 $response =  $chat->markdownV2(
"Thank you for your message.  We have sent it via email to the office at the Cookery School and they will reply as soon as they can.
If it is an emergency, please phone Toby on  353 87 6200000 or Jim on  353 87 2300000 "
            )->send();
````

**Comment by tobya** [comment]
Ok. I figured this out, I need to escape `.` also !!!!!

Full list here

https://core.telegram.org/bots/api#markdownv2-style

**Reply by fabio-ivona** [reply]
Glad you found a solution and pasted it here for future reference, thanks!

**Reply by tobya** [reply]
Thanks.  Would it be possible to raise an error if markdown is found to be invalid by Telegram?  Maybe it is and I just havent set the setting.

**Reply by fabio-ivona** [reply]
we will investigate on it
